src/librepgp/stream-armor.cpp

```diff
@@ -1006,23 +1028,8 @@ rnp_dearmor_source(pgp_source_t *src, pgp_dest_t *dst)
         RNP_LOG("source is not armored data");
         return RNP_ERROR_BAD_FORMAT;
     }
-
     /* Reading data from armored source and writing it to the output */
-    while (!armorsrc.eof) {
-        if (!src_read(&armorsrc, readbuf, PGP_INPUT_CACHE_SIZE, &read)) {
-            res = RNP_ERROR_GENERIC;
-            break;
-        }
-        if (!read) {
-            continue;
-        }
-        dst_write(dst, readbuf, read);
-        if (dst->werr) {
-            RNP_LOG("failed to output data");
-            res = RNP_ERROR_WRITE;
-            break;
-        }
-    }
+    res = dst_write_src(&armorsrc, dst);
```

Comment:
Previously we were logging an error if this operation fails, so we should log it here as well, so in case of error it would be easier to detect the problem.

Reply:
And, some possible refactoring - as `armorsrc` is the only resource freed after goto to finish, we may safely return if `init_armored_src` failed, removing the goto label.